### PR TITLE
Consolidation of old and new installation

### PR DESCRIPTION
**Pull request by taimir**
Merged the new option to install k8s as a service on the host
into the old provisioning scripts.

So right now you should be able to start the `master.sh` and `minion.sh` scripts with a parameter value either `hyperkube` or `service`, corresponding to the two installation options.

I'm not really sure if we want to support two restart policies for hyperkube, @cheld what is the idea?

### CODE REVIEW

File: master.sh [hunk 1]
```diff
@@ -131,7 +125,49 @@ detect_lsb() {
     esac
 }
 
-start_k8s(){
+configure_docker(){
```
Comment (taimir) [inline master.sh:136]: I just moved this function a bit to the top of the file, no changes here.

File: master.sh [hunk 2]
```diff
@@ -40,13 +40,7 @@ init() {
 	fi
 
 	# Set working mode
-	if [[ -n ${START_MODE} && ${START_MODE} == "install" ]]; then
-		echo "Start k8s cluster in installation mode"
 		RESTART_POLICY="always"
```
Comment (taimir) [inline master.sh:46]: @zreigz @cheld I thought the start mode for the scripts was supposed to be either as a `systemd service` or as a `hyperkube` container. Do we additionally want to support different restart policies for `hyperkube`? If so I can easily revert my changes, but we'll need one more additional parameter passed.

**Comment by taimir** [comment]
Tested it on the `kube-deploy` setup, so far it seems to work fine.

File: master.sh (hunk 2 above)
Comment (cheld) [inline master.sh:46]: currently, we still have a few container outside of kublet: etcd, flannel, proxy.

We change the start mode to the use case accordingly .

Of course the goal would be to move them into kublet

File: master.sh [hunk 3]
```diff
@@ -211,48 +260,26 @@ start_k8s(){
...
+        --net=host \
+        --pid=host \
+        --privileged \
+        taimir93/hyperkube-amd64:v1.3.0-beta.2 \
+        /setup/install.sh master localhost ${MASTER_IP} ${PAUSE_IMAGE}
+
+    sleep 3
+
+    echo "Running kubelet ..."
               systemctl daemon-reload
```
Comment (cheld) [inline master.sh]: small issue: too much tabbing space...

File: master.sh (hunk 2 above)
Comment (cheld) [inline master.sh:46]: Please keep this code

File: worker.sh [hunk 4]
```diff
@@ -195,57 +244,39 @@ start_k8s() {
...
-              DOCKER_CONF=$(systemctl cat docker | head -1 | awk '{print $2}')
-              sed -i.bak 's/^\(MountFlags=\).*/\1shared/' $DOCKER_CONF
+start_service_k8s(){
+		echo "Installing k8s as a service ..."
+		docker run -d \
+		    --volume=/:/hostfs:rw \
+		    --net=host \
+		    --pid=host \
+		    --privileged \
+		    taimir93/hyperkube-amd64:v1.3.0-beta.2 \
```
Comment (cheld) [inline worker.sh]: you can use fest (fujitsu est) orgainization. register on docker hub, add your company email to your profile and I grant you access

File: worker.sh [hunk 5]
```diff
@@ -195,57 +244,39 @@ start_k8s() {
...
+		    /setup/install.sh minion ${MASTER_IP} ${NODE_IP} ${PAUSE_IMAGE}
```
Comment (cheld) [inline worker.sh]: we should use default pause image or upload to fest organization

File: master.sh [hunk 6]
```diff
@@ -178,7 +214,20 @@ start_k8s(){
...
+      start_hyperkube_k8s
+    elif [ "$1" == "service" ]; then
+      start_service_k8s
+    else
+      echo "You should specify how to start kubelet: as a service or in hyperkube."
+      exit 1
+    fi
+}
+
+start_hyperkube_k8s(){
```
Comment (cheld) [inline master.sh]: Small comment: I would change the name to indicate close relations:
start_k8s
start_k8s_hyperkube   or   start_k8s_as_hyperkube 
start_k8s_service         or   start_k8s_as_service

File: master.sh (hunk 2 above)
Comment (cheld) [inline master.sh:46]: Ok let me explain. With your change, after reboot, k8s is running independent of the mode "run" or "install". Only implementation detail (the way it is installed) changed. 

My intention was rather to support two use cases: 1. the user wants to run k8s without installation. 2. the user wants to install k8s. The reboot behavior should be set accordingly

Makes sense?

**Comment by cheld** [comment]
Very nice! A few small comments

File: master.sh (hunk 2 above)
Comment (taimir) [inline master.sh:46]: OK, this sounds reasonable. My question was rather: in the `install` case, should I replace the existing `hyperkube + restart_policy=always` with the new `kubelet as service` installation, or still keep them both for some reason? I don't see the benefit of having both for the `install` right now, maybe you could clarify this.

Personally I would keep `hyperkube` for the `run once` option, and start kubelet as a service for the `install` option.

**Comment by cheld** [comment]
LGTM

**Pull request merged**